python/tvm/hybrid/parser.py

```diff
@@ -0,0 +1,342 @@
+
+
+    def visit_BinOp(self, node):
+        lhs = self.visit(node.left)
+        rhs = self.visit(node.right)
+        return HybridParser._binop_maker[type(node.op)](lhs, rhs)
+
+
+    def visit_Call(self, node):
+        # Yet, no function pointer supported
+        if not isinstance(node.func, ast.Name):
+            raise ValueError("Only id-function function call is supported so far!")
+        func_id = node.func.id
+        n = len(node.args)
+        if func_id in LOOP_INTRIN.keys() and func_id != 'bind':
+            if n == 1:
+                low, ext = _api.const(0, dtype='int32'), self.visit(node.args[0])
+            else:
+                if n != 2:
+                    raise ValueError("A loop intrinsic should only have 1 or 2 arguments!")
+                low, ext = self.visit(node.args[0]), self.visit(node.args[1])
+            if not _ir_pass.Equal(low, _api.const(0, dtype='int32')):
+                ext = ext - low
+            for_type = LOOP_INTRIN[func_id]
+            iter_var = None
+            return iter_var, low, ext, for_type
+        elif func_id == 'bind':
+            if n != 2:
+                raise ValueError("A loop bind should only have 2 arguments!")
+            if not isinstance(node.args[0], ast.Str):
+                raise ValueError("A loop bind's first argument should be a string!")
+            _vn = node.args[0].s
+            iter_var = thread_axis(node.args[0].s)
+            low, ext = _api.const(0, dtype='int32'), self.visit(node.args[1])
+            for_type = None
+            return iter_var, low, ext, for_type
+        elif func_id in MATH_INTRIN:
+            return getattr(intrin, func_id)(*[self.visit(arg) for arg in node.args])
+        elif func_id == 'allocate':
+            #TODO: Support it later!
+            return make_nop()
+        else:
+            raise ValueError("Function call not supported yet!")
+
+
+    def visit_For(self, node):
+        iter_var, low, ext, for_type = self.visit(node.iter)
+        if not isinstance(node.target, ast.Name):
+            raise ValueError("The loop iterator should be a variable!")
+        _name = node.target.id
+        if iter_var is None:
+            if for_type is None:
+                raise ValueError("The loop bind function parse error!")
+            iter_var = _api.var(_name)
+            self.loops_above[_name] = iter_var
+        else:
+            if for_type is not None:
+                raise ValueError("The loop iterating function parse error!")
+            self.loops_above[_name] = iter_var.var
+        _body = list_to_block(self.visit, node.body)
+        _body = self.wrap_up_realize(node, _body)
+        if for_type is None:
+            res = _make.AttrStmt(iter_var, 'thread_extent', ext, _body)
+        else:
+            res = _make.For(iter_var, low, ext, for_type, 0, _body)
+        self.loops_above.pop(_name)
+        return res
+
+
+def parse_python(src, args):
```

Comment:
is this function necessary? Can we just do  and inline it into parse?
```python
HybridParser parser(src, args)
return parser.parse(root)
```

Reply:
I just want to make the code style uniform, each IR pass has its own helper function. Just like that is in Halide style.